File: diagonal-difference.py
def diagonalDifference(arr):
    
    a_sum = 0
    b_sum = 0
    
    a_index = 0
    b_index = len(arr) - 1
    
    for row in arr:
        a_sum += row[a_index]
        b_sum += row[b_index]
        
        a_index += 1
        b_index -=1
                
    print(f'Final A Sum: {a_sum}')
    print(f'Final B Sum: {b_sum}')           
    return abs(a_sum - b_sum)

File: diagonal-difference.js
function diagonalDifference(arr) {
    let [ a_sum, b_sum, a_index, b_index ] = [0, 0, 0, arr.length - 1]

    while (a_index < arr.length && b_index >= 0) {
        a_sum += arr[a_index][a_index]; // Access primary diagonal
        b_sum += arr[a_index][b_index]; // Access secondary diagonal

        console.log(`Current A Sum: ${a_sum}`);
        console.log(`Current B Sum: ${b_sum}`);

        a_index++;
        b_index--;
    }

    return Math.abs(a_sum - b_sum);
}
